drivers/i2s/i2s_mcux_sai.c

```diff
@@ -434,6 +420,34 @@ static void get_mclk_rate(const struct device *dev, uint32_t *mclk)
 	*mclk = rate;
 }
 
+static void enable_mclk_direction(const struct device *dev, bool dir)
+{
+	const struct i2s_mcux_config *dev_cfg = dev->config;
+
+#ifdef CONFIG_I2S_HAS_PLL_SETTING
```

Comment:
Not a blocking comment, just a general concern- we probably need to consider a more generic way to set the MCLK pin direction here. The block here seems to be specific to the RT1xxx series parts, but configuring the MCLK pin is an operation generic across I2S peripherals.

Perhaps we need an SOC specific function to handle this? @DerekSnell are you aware, does the MCLK direction setting end up being very SOC-specific? IE would it make the most sense to implement it as SOC specific code?

Reply:
Hi @danieldegrasse and @mcuxted ,
Sorry, I do not have experience with these SAI SOC differences.  I think it would be best to refer to the MCUXpresso SDK driver examples for the different SOCs.  It appears these are the MCUs supported in Zephyr today with the SAI: RT10xx, RT11xx, RT700, MCXC, MCXN, Kinetis K like K22F12.

Reply:
Hi @DerekSnell, I have updated it according to the latest code, adding pinmuxes to uniformly modify the direction of mclk